### DISCUSSION TITLE
How to fetch by useQuery on page load.

### DISCUSSION THREAD

**Discussion by theMistletoe** [TanStack/query#4536] [Q&A] [answered]
I want to fetch data on page loaded, like this we used to wrote code...

```ts
useEffect(() => {
    axios.get(
      "https://jsonplaceholder.typicode.com/posts"
    );
  }, [])
```

I prepare sample code. https://github.com/theMistletoe/tanstack-vite-sample/tree/main
I supposed to think when I visit on this sample page, I can watch fetched data, but I watch "loading" message...
I checked dev tools, it doesn't call any XHR request.

maybe, I make a mistake.
give me some advice.

This sample is similar to my app’s working.
https://tanstack.com/query/v4/docs/examples/react/basic
On first view, I think I can watch some posts, but it’s eternally loading.

**Comment by TkDodo** [comment, marked as answer]
I don't understand the question. your code looks fine ...

**Reply by theMistletoe** [reply]
uhh…

> https://tanstack.com/query/v4/docs/examples/react/basic
On first view, I think I can watch some posts, but it’s eternally loading.

this problem is solved, now.
Its sample is working correctlly.
Maybe, this was my local envirronment’s problem, ex browser.

thank you for your review!
